lizmap/modules/lizmap/lib/Request/Proxy.php

```diff
@@ -500,87 +337,30 @@ protected static function logRequestIfError($httpCode, $url)
-     *                                   it is ignored if $options is an array.
-     * @param string|string[]   $method  deprecated. the http method.
-     *                                   it is ignored if $options is an array.
-     *
-     * @return array{0: string, 1: string, 2: int} Array containing data (0: string), mime type (1: string) and HTTP code (2: int)
-     */
-    public static function getRemoteData($url, $options = null, $debug = null, $method = 'get')
-    {
-        $options = self::buildOptions($options, $method, $debug);
-        list($url, $options) = self::buildHeaders($url, $options);
-
-        // check is the env variable is set
-        if (getenv('ECHO_OGC_ORIGINAL_REQUEST')) {
-            // did the request has to be echoed ?
-            if (self::hasEchoInBody($options['body'])) {
-                $content = self::getEchoFromRequest($url, $options['body']);
-
-                // We do not perform the request, but return the content previously logged
-                return array(
-                    $content,
-                    'text/json',
-                    200,
-                );
-            }
-            // All requests are logged
-            self::logRequestToEcho($url, $options['body']);
-        }
-
-        // Proxy http backend : use curl or file_get_contents
-        if (extension_loaded('curl') && $options['proxyHttpBackend'] != 'php') {
-            // With curl
-            $curlRequest = self::curlProxy($url, $options);
-            self::logRequestIfError($curlRequest[2], $url);
-
-            return $curlRequest;
-        }
-        // With file_get_contents
-        $request = self::fileProxy($url, $options);
-        self::logRequestIfError($request[2], $url);
-
-        return $request;
-    }
-
-    /**
-     * Sends a request, and return the body response as a stream.
+     * Sends a request with Guzzle.
      *
      * @param string     $url     url of the remote data to fetch
      * @param null|array $options list of options for the http request.
      *                            Option items can be: "method", "referer", "proxyHttpBackend",
-     *                            "headers" (array of headers strings), "body", "debug".
+     *                            "headers" (array of headers strings), "body", "debug", stream.
      *
-     * @return ProxyResponse
+     * @return ResponseInterface
      */
-    public static function getRemoteDataAsStream($url, $options = null)
+    protected static function sendRequest($url, $options)
     {
-        $options = self::buildOptions($options, 'get', null);
         list($url, $options) = self::buildHeaders($url, $options);
+
         // check is the env variable is set
         if (getenv('ECHO_OGC_ORIGINAL_REQUEST')) {
             // did the request has to be echoed ?
             if (self::hasEchoInBody($options['body'])) {
                 $content = self::getEchoFromRequest($url, $options['body']);
```

Comment:
the `ECHO_OGC_ORIGINAL_REQUEST` part can be refactored as middleWare / handler for [guzzle](https://docs.guzzlephp.org/en/stable/handlers-and-middleware.html#handlers) , improvement for another PR ?

Reply:
yes sure